<a href="https://colab.research.google.com/github/ssook1222/bigdata-analysist-certificate/blob/master/bigdata-analysist-certificate/task1/task1_10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 센서데이터 동작유형 분류 데이터

In [1]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/muscle/train.csv")
df.head()

motion_0  motion_1  motion_2  motion_3  motion_4  motion_5  motion_6  \
0      -6.0       8.0       5.0       6.0     -12.0     -27.0       5.0   
1     -32.0      -3.0      -4.0      -5.0      -1.0      16.0       0.0   
2      35.0      -2.0      -1.0       1.0     -25.0     -16.0      -3.0   
3     -15.0      -6.0      -3.0      -5.0      27.0      37.0      -2.0   
4      60.0       3.0       0.0      -7.0      -6.0     -10.0      -6.0   

   motion_7  motion_8  motion_9  ...  motion_55  motion_56  motion_57  \
0      24.0      -8.0      -5.0  ...       -1.0       -3.0       -6.0   
1      -5.0      42.0       2.0  ...      -17.0      -12.0       -1.0   
2     -20.0      10.0       1.0  ...        6.0      -65.0        6.0   
3      -2.0      19.0       5.0  ...        2.0       11.0        0.0   
4       7.0     -11.0      -4.0  ...      -12.0       17.0       -3.0   

   motion_58  motion_59  motion_60  motion_61  motion_62  motion_63  pose  
0       -9.0       -7.0       36.0       68.0        3.0       -3.0     1  
1       -1.0        0.0      -26.0      -11.0       -4.0       -9.0     1  
2      -10.0      -31.0      -19.0       -3.0      -47.0      -17.0     0  
3       -2.0       -4.0       -7.0       15.0        3.0       14.0     1  
4        1.0        2.0       -2.0       20.0       18.0       10.0     0  

[5 rows x 65 columns]

### Question 100

pose값에 따른 각 motion컬럼의 중간값의 가장 큰 차이를 보이는 motion컬럼은 어디이며 그값은?

In [2]:
t = df.groupby('pose').median().T

abs(t[0] - t[1]).sort_values(ascending = False).head(2).index

Index(['motion_54', 'motion_62'], dtype='object')

T를 붙이면 행렬 전치 가능

## 현대 차량 가격 분류문제 데이터

In [3]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/hyundai/train.csv")
df.head()

model  year  price transmission  mileage fuelType  tax(£)   mpg  \
0   Tucson  2016  12795       Manual    36263   Diesel      30  61.7   
1      I10  2012   3995       Manual    22089   Petrol      20  61.4   
2      I30  2019  24000       Manual      555   Petrol     145  34.0   
3   Tucson  2017  12995       Manual    32147   Diesel      30  61.7   
4   Tucson  2018  14991    Semi-Auto    32217   Diesel     150  57.6   

   engineSize  
0         1.7  
1         1.2  
2         2.0  
3         1.7  
4         1.7

### Question 101

정보(row수)가 가장 많은 상위 3차종의 price값의 각 평균값은?

In [4]:
pd.DataFrame(df[(df['model'].isin(df[['model']].groupby('model').size().sort_values(ascending = False).head(3).index))].groupby('model').mean().price)

<ipython-input-4-18de5c60b475>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  pd.DataFrame(df[(df['model'].isin(df[['model']].groupby('model').size().sort_values(ascending = False).head(3).index))].groupby('model').mean().price)


price
model                
 I10      7646.137891
 I30     11651.821759
 Tucson  15805.441373

## 당뇨여부판단 데이터

In [5]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/diabetes/train.csv")
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            3      102             74              0        0  29.5   
1            2      144             58             33      135  31.6   
2            5      136             82              0        0   0.0   
3           13      145             82             19      110  22.2   
4            1      117             60             23      106  33.8   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.121   32        0  
1                     0.422   25        1  
2                     0.640   69        0  
3                     0.245   57        0  
4                     0.466   27        0

### Question 102

Outcome 값에 따른 각 그룹의 각 컬럼의 평균 차이를 구하여라

In [6]:
df.groupby('Outcome').mean().diff()[1:]

Pregnancies    Glucose  BloodPressure  SkinThickness    Insulin  \
Outcome                                                                    
1           1.574159  29.943995       3.202079       2.952033  33.398645   

              BMI  DiabetesPedigreeFunction       Age  
Outcome                                                
1        4.541437                  0.143646  5.866939

## 넷플릭스 주식 데이터

In [7]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/nflx/NFLX.csv")
df.head()

Date      Open      High       Low     Close  Adj Close     Volume
0  2002-05-23  1.156429  1.242857  1.145714  1.196429   1.196429  104790000
1  2002-05-24  1.214286  1.225000  1.197143  1.210000   1.210000   11104800
2  2002-05-28  1.213571  1.232143  1.157143  1.157143   1.157143    6609400
3  2002-05-29  1.164286  1.164286  1.085714  1.103571   1.103571    6757800
4  2002-05-30  1.107857  1.107857  1.071429  1.071429   1.071429   10154200

### Question 103

매년 5월달의 open가격의 평균값을 데이터 프레임으로 표현하라

In [8]:
import datetime as dt

In [21]:
df['Date'] = pd.to_datetime(df.Date)
df103 = df[(df['Date'].dt.month == 5)]
df103['Date'] = pd.to_datetime(df103.Date)
df103['Date'] = df103.Date.dt.strftime('%Y-%m')
df103[['Date', 'Open']].groupby('Date').mean()

<ipython-input-21-50725187d942>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df103['Date'] = pd.to_datetime(df103.Date)
<ipython-input-21-50725187d942>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df103['Date'] = df103.Date.dt.strftime('%Y-%m')


Open
Date               
2002-05    1.155833
2003-05    1.641497
2004-05    4.261143
2005-05    1.951905
2006-05    4.163571
2007-05    3.159351
2008-05    4.435034
2009-05    5.790571
2010-05   14.417071
2011-05   34.650272
2012-05   10.523247
2013-05   31.936429
2014-05   51.121292
2015-05   85.057429
2016-05   92.705715
2017-05  158.255455
2018-05  329.779541
2019-05  359.664548
2020-05  433.880499
2021-05  496.923996

`strftime('%Y-%m'))` : 년도 - 월로 이렇게 DateTime 원하는 대로 양식 세팅 가능

## nba 선수 능력치 데이터

In [22]:
pd.set_option('display.max_columns',50)
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/nba/nba.csv",encoding='latin',sep=';')
df.head()

Rk             Player Pos  Age   Tm   G  GS    MP   FG   FGA    FG%   3P  \
0   1   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8   
1   2       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0   
2   3        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0   
3   4       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2   
4   5  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3   

   3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  AST  \
0  2.1  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5  1.1   
1  0.0  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0  3.4   
2  0.1  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1  3.4   
3  1.5  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7  0.7   
4  1.0  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5  0.9   

   STL  BLK  TOV   PF   PTS  
0  0.5  0.6  1.2  2.1   9.1  
1  0.9  0.8  1.5  2.0   6.9  
2  1.4  0.8  2.6  3.1  19.1  
3  0.2  0.3  0.5  1.1   4.1  
4  0.3  1.0  0.9  1.7  12.9

### Question 104

Tm 컬럼은 각 팀의 이름을 의미한다. TOR팀의 평균나이를 소수 4째 자리까지 구하여라

In [25]:
round(df[(df.Tm == 'TOR')].mean().Age, 4)

<ipython-input-25-cb46ef44a9bc>:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  round(df[(df.Tm == 'TOR')].mean().Age, 4)


24.8696

### Question 105

Pos 컬럼은 포지션을 의미한다. 전체 선수 중 최소나이대의 선수들을 필터하고 그들 중 가장 많은 포지션은 무엇인지 확인하라

In [32]:
df[(df.Age!=df.Age.min)].Pos.value_counts().head(1).index

Index(['SG'], dtype='object')

### Question 106

선수들의 이름은 first_name+ 공백 + last_name으로 이루어져 있다. 가장 많은 first_name은 무엇이며 몇 회 발생하는지 확인하라

In [39]:
def name(x):
  return x.split()[0]

df['First'] = df.Player.apply(name)
df.First.value_counts().head(1)

Justin    14
Name: First, dtype: int64

### Question 107

PTS컬럼은 경기당 평균득점수 이다. 각포지션별로 경기당 평균득점수의 평균을 구하여라

In [41]:
df[['PTS', 'Pos']].groupby('Pos', as_index = False).mean()

Pos        PTS
0       C   7.690769
1    C-PF  12.850000
2      PF   7.737500
3   PF-SF   8.500000
4      PG   8.325161
5   PG-SG  22.000000
6      SF   7.270253
7   SF-SG   8.660000
8      SG   7.810553
9   SG-PG   9.525000
10  SG-SF   7.360000

### Question 109

G컬럼은 참여한 경기의 숫자이다. 각 팀별로 가장 높은 경기참여 수를 가진 선수들의 경기 참여 숫자의 평균을 구하여라

In [52]:
df[['G', 'Tm']].groupby('Tm', as_index = False).max().mean().G

<ipython-input-52-63f6fc6a89f7>:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df[['G', 'Tm']].groupby('Tm', as_index = False).max().mean().G


77.51612903225806

### Question 110

Tm의 값이 MIA이며 Pos는 C또는 PF인 선수의 MP값의 평균은?

In [58]:
df[(df.Tm == 'MIA') & ((df.Pos == 'C') | (df.Pos == 'PF'))].MP.mean()

16.7875

### Question 111

전체 데이터중 G의 평균값의 1.5배 이상인 데이터들만 추출했을때 3P값의 평균은?

In [80]:
df[(df.G >= df.G.mean() * 1.5)]['3P'].mean()

1.3853658536585365

### Question 112

Age의 평균 이상인 그룹과 평균 미만인 그룹간의 G값의 평균의 차이는?

In [87]:
df112_up = df[(df.Age >= df.Age.mean())]
df112_down = df[(df.Age < df.Age.mean())]

df112_up.G.mean() - df112_down.G.mean()

3.787674551781862

### Question 113

평균나이가 가장 젊은 팀은 어디인가

In [98]:
df.groupby('Tm')['Age'].mean().sort_values().head(1).index[0]

'MEM'

### Question 114

Pos그룹별 평균 MP값을 구하여라

In [103]:
df114 = df.groupby('Pos').mean()
df114['MP']

<ipython-input-103-b378e9568f99>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df114 = df.groupby('Pos').mean()


Pos
C        16.990000
C-PF     25.350000
PF       17.937500
PF-SF    27.300000
PG       19.547742
PG-SG    37.200000
SF       17.514557
SF-SG    20.340000
SG       18.554271
SG-PG    22.950000
SG-SF    19.620000
Name: MP, dtype: float64